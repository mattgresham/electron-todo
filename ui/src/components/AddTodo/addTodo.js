import React from 'react';
import PropTypes from 'prop-types';
import styles from './addTodo.module.css';

const AddTodo = ({ addTodo, todoText, setTodos, setTodoText }) => (
  <form
    className={styles.container}
    onSubmit={e => {
      e.preventDefault();
      addTodo(todoText, setTodos, setTodoText);
    }}
  >
    <input
      className={styles.textField}
      onChange={e => setTodoText(e.target.value)}
      placeholder="Enter something to do..."
      type="text"
      value={todoText}
    />
    <button className={styles.addButton} disabled={!todoText} type="submit">
      Add
    </button>
  </form>
);

AddTodo.propTypes = {
  addTodo: PropTypes.func.isRequired,
  todoText: PropTypes.string.isRequired,
  setTodos: PropTypes.func.isRequired,
  setTodoText: PropTypes.func.isRequired
};

export default AddTodo;
