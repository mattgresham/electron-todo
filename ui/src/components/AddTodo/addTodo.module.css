.container {
  display: flex;
  justify-content: space-between;
  margin-top: 0.5rem;
}

.textField {
  border-radius: 0.25rem;
  border: 1px solid #d3d3d3;
  font-size: 2rem;
  margin-right: 1rem;
  padding: 0.25rem;
  width: 100%;
}

.textField::placeholder {
  opacity: 0.5;
}

.addButton {
  background: none;
  border: 1px solid #d3d3d3;
  border-radius: 0.25rem;
  font-size: 1.5rem;
  padding: 0.5rem;
}
