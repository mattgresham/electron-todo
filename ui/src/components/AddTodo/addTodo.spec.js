import React from 'react';
import { render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import AddTodo from '.';

configure({ adapter: new Adapter() });

it(`AddTodo renders with default props`, () => {
  const tree = render(
    <AddTodo
      addTodo={() => {}}
      setTodos={() => {}}
      setTodoText={() => {}}
      todoText=""
    />
  );
  expect(toJson(tree)).toMatchSnapshot();
});

it(`AddTodo renders with a text prop`, () => {
  const tree = render(
    <AddTodo
      addTodo={() => {}}
      setTodos={() => {}}
      setTodoText={() => {}}
      todoText="Todo"
    />
  );
  expect(toJson(tree)).toMatchSnapshot();
});
