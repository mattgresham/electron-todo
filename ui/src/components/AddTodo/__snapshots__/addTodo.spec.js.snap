// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo renders with a text prop 1`] = `
<form
  class="container"
>
  <input
    class="textField"
    placeholder="Enter something to do..."
    type="text"
    value="Todo"
  />
  <button
    class="addButton"
    type="submit"
  >
    Add
  </button>
</form>
`;

exports[`AddTodo renders with default props 1`] = `
<form
  class="container"
>
  <input
    class="textField"
    placeholder="Enter something to do..."
    type="text"
    value=""
  />
  <button
    class="addButton"
    disabled=""
    type="submit"
  >
    Add
  </button>
</form>
`;
